salt/modules/cohesity.py

```diff
@@ -0,0 +1,532 @@
+    from cohesity_management_sdk.cohesity_client import CohesityClient
+    from cohesity_management_sdk.exceptions.api_exception import APIException
+    from cohesity_management_sdk.models.cancel_protection_job_run_param import (
+        CancelProtectionJobRunParam,
+    )
+    from cohesity_management_sdk.models.change_protection_job_state_param import (
+        ChangeProtectionJobStateParam,
+    )
+    from cohesity_management_sdk.models.delete_protection_job_param import (
+        DeleteProtectionJobParam,
+    )
+    from cohesity_management_sdk.models.environment_register_protection_source_parameters_enum import (
+        EnvironmentRegisterProtectionSourceParametersEnum as env_enum,
+    )
+    from cohesity_management_sdk.models.protection_job_request_body import (
+        ProtectionJobRequestBody,
+    )
+    from cohesity_management_sdk.models.recover_task_request import RecoverTaskRequest
+    from cohesity_management_sdk.models.register_protection_source_parameters import (
+        RegisterProtectionSourceParameters,
+    )
+    from cohesity_management_sdk.models.restore_object_details import (
+        RestoreObjectDetails,
+    )
+    from cohesity_management_sdk.models.run_protection_job_param import (
+        RunProtectionJobParam,
+    )
+    from cohesity_management_sdk.models.universal_id import UniversalId
+    from cohesity_management_sdk.models.update_protection_jobs_state_request_body import (
+        UpdateProtectionJobsStateRequestBody,
+    )
+    from cohesity_management_sdk.models.vmware_restore_parameters import (
+        VmwareRestoreParameters,
+    )
+
+    HAS_LIBS = True
+
+    # Update the user-agent.
+    BaseController.global_headers['user-agent'] = 'cohesity-salt/1.0.0'
+except ImportError as err:
+    HAS_LIBS = False
+
+log = logging.getLogger(__name__)
+__virtualname__ = "cohesity"
+
+
+def _get_client():
+    context_key = "{}.cohesity_client".format(__name__)
+    if context_key in __context__:
+        return __context__[context_key]
+    cohesity_config = __opts__.get("cohesity_config", {})
+    cluster_vip = cohesity_config.get("cluster_vip", "")
+    c_username = cohesity_config.get("username", "")
+    c_password = cohesity_config.get("password", "")
+    c_domain = cohesity_config.get("domain", "")
+    cohesity_client = CohesityClient(
+        cluster_vip=cluster_vip,
+        username=c_username,
+        password=c_password,
+        domain=c_domain,
+    )
+    __context__[context_key] = cohesity_client
+    return cohesity_client
+
+
+def __virtual__():
+    if HAS_LIBS:
+        return __virtualname__
+    return False, "Could not import the Cohesity SDK modules"
+
+
+def _get_sd_id(name):
+    """
+    Function to fetch storage domain available in the cluster.
+
+    : return: storage domain id.
+    """
+    cohesity_client = _get_client()
+    log.info("Getting storage domain with name %s" % name)
```

Comment:
```suggestion
    log.info("Getting storage domain with name %s", name)
```
Sorry, this is how log calls should be done.

Reply:
Okay Thank you,